addons/binding/org.openhab.binding.ihc/src/main/java/org/openhab/binding/ihc/internal/ws/projectfile/ProjectFileUtils.java

```diff
@@ -0,0 +1,169 @@
+     */
+    public static void saveToFile(String filePath, byte[] data) throws IhcExecption {
+        try {
+            FileUtils.writeByteArrayToFile(new File(filePath), data);
+        } catch (IOException e) {
+            throw new IhcExecption(e);
+        }
+    }
+
+    /**
+     * Convert bytes to XML document.
+     *
+     * @return XML document or null if conversion fails.
+     */
+    public static Document converteBytesToDocument(byte[] data) {
+        DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
+        factory.setNamespaceAware(true);
+        try {
+            DocumentBuilder builder = factory.newDocumentBuilder();
+            return builder.parse(new ByteArrayInputStream(data));
+        } catch (ParserConfigurationException | SAXException | IOException e) {
+            LOGGER.warn("Error occured when trying to convert data to XML, reason {}", e.getMessage());
+        }
+        return null;
+    }
+
+    /**
+     * Compare XML document header information to project info.
+     *
+     * @return true if information is equal and false if not.
+     */
+    public static boolean projectEqualsToControllerProject(Document projectfile, WSProjectInfo projectInfo) {
+        if (projectInfo != null) {
+            try {
+                NodeList nodes = projectfile.getElementsByTagName("modified");
+                if (nodes.getLength() == 1) {
+                    Element node = (Element) nodes.item(0);
+                    int year = Integer.parseInt(node.getAttribute("year"));
+                    int month = Integer.parseInt(node.getAttribute("month"));
+                    int day = Integer.parseInt(node.getAttribute("day"));
+                    int hour = Integer.parseInt(node.getAttribute("hour"));
+                    int minute = Integer.parseInt(node.getAttribute("minute"));
+
+                    LOGGER.debug("Project file from file, date: {}.{}.{} {}:{}", year, month, day, hour, minute);
+                    LOGGER.debug("Project file in controller, date: {}.{}.{} {}:{}",
+                            projectInfo.getLastmodified().getYear(),
+                            projectInfo.getLastmodified().getMonthWithJanuaryAsOne(),
+                            projectInfo.getLastmodified().getDay(), projectInfo.getLastmodified().getHours(),
+                            projectInfo.getLastmodified().getMinutes());
+
+                    if (projectInfo.getLastmodified().getYear() == year
+                            && projectInfo.getLastmodified().getMonthWithJanuaryAsOne() == month
+                            && projectInfo.getLastmodified().getDay() == day
+                            && projectInfo.getLastmodified().getHours() == hour
+                            && projectInfo.getLastmodified().getMinutes() == minute) {
+                        return true;
+                    }
+                }
+            } catch (Exception e) {
+                // do nothing, but return false
```

Comment:
Catching Exception and logging at any level is imho a really bad idea

Reply:
In general level I agree, but...

when you don't have documentation and API available, but I have to reverse engineer the the whole protocol, you basically don't know what could go wrong. I have made lot of guesses and assumptions. There are multiple versions of the IHC controller and also many software versions, which have slight differences. Of course you could use lot of time to understand what could go wrong in every case, but I just don't have that luxury. E.g. this feature is fully secondary feature of the binding. It just try to make initialisation faster (method could always return false). In the end I need to support the other users as well, so small wrong assumptions can easily cause that binding is useless for other users or if problem occurs very rarely, it can be challenging to find the root cause as I don't have same version of the hw. 

Anyhow, I have added more inline comments to code.

Reply:
No. If you worry about unexpected exceptions you need to catch `RuntimeException` those are the unchecked exceptions you want to catch. And always log! Because if the user hits an exception and it isn't logged it just might mean the binding won't work and the user will be left in the dark. When you than want to help the user you have also nothing to fall back on, and since you probably don't have the same hardware can't reproduce it yourself. So if debug or trace logging would have been done you can pinpoint the problem. I always use defensive programming and would set the thing offline with the error message. This could mean that theoretically the thing still might have worked. But it will give the best feedback loop. Because the user directly knows something is wrong and will very likely report about it and provide you with an opportunity to improve the binding.

Reply:
I fully agree about the logging and that I have added as well. I just replied to Martins comment.

> I always use defensive programming and would set the thing offline with the error message.

This is exactly what I want to avoid when you meet unexpected errors in inessential/secondary features. Without that user can't use the binding at all before I or others will make a fix for that, which could take weeks/months/years depending of the maintainers.

Reply:
> I fully agree about the logging and that I have added as well

Ok I missed that part from your comment.  Please do change the catch to `RuntimeException`.

> This is exactly what I want to avoid when you meet unexpected errors 

That's fine, it's a personal choice of the developer. I understood that from your previous comment. I wanted to add my own personal view :smile:

Reply:
It didn't help that I forgot a NOT here 😭 

> Catching Exception and logging at any level is imho a really bad idea

I meant 

> Catching Exception and NOT logging at any level is imho a really bad idea

and the focus of my comment was meant on the logging, so it resulted in the correct actions.